src/test/end.t.sol

```diff
@@ -483,6 +504,126 @@ contract EndTest is DSTest {
         assertEq(balanceOf("gold", address(gold.gemA)), 0);
     }
 
+    // -- Scenario where there is one collateralised CDP
+    // -- undergoing auction at the time of cage
+    function test_cage_snip() public {
+        Ilk memory gold = init_collateral("gold");
+
+        Usr ali = new Usr(vat, end);
+
+        // Make a CDP:
+        address urn1 = address(ali);
+        gold.gemA.join(urn1, 10 ether);
+        ali.frob("gold", urn1, urn1, urn1, 10 ether, 15 ether);
+        (uint ink1, uint art1) = vat.urns("gold", urn1); // CDP before liquidation
+        (, uint rate,,,) = vat.ilks("gold");
+        assertEq(vat.gem("gold", urn1), 0);
+        assertEq(rate, ray(1 ether));
+        assertEq(ink1, 10 ether);
+        assertEq(art1, 15 ether);
+
+        vat.file("gold", "spot", ray(1 ether)); // Now unsafe
+
+        uint id = dog.bark("gold", urn1);                   
+        (, uint256 tab, uint256 lot,,,) = gold.clip.sales(id);  
+        assertEq(tab, rad(16.5 ether));             
+        assertEq(dog.Dirt(), tab);
+        assertEq(lot, ink1);
+
+        {
+            (uint ink2, uint art2) = vat.urns("gold", urn1); // CDP after liquidation
+            assertEq(ink2, 0);
+            assertEq(art2, 0);
+        }
+
+        // Collateral price is $5
+        gold.pip.poke(bytes32(5 * WAD));
+        spot.poke("gold");
+        end.cage();
+        end.cage("gold");
+        assertEq(end.tag("gold"), ray(0.2 ether)); // par / price = collateral per DAI
+
+        assertEq(vat.gem("gold", address(gold.clip)), lot); // From grab in dog.bark()
+        assertEq(vat.sin(address(vow)),       art1 * rate); // From grab in dog.bark()
+        assertEq(vat.vice(),                  art1 * rate); // From grab in dog.bark()
+        assertEq(vat.debt(),                  art1 * rate); // From frob
+        assertEq(vat.dai(address(vow)),                 0); // vat.suck() hasn't been called 
+
+        end.snip("gold", id);
+
+        assertEq(dog.Dirt(),                            0); // From clip.yank()
+        assertEq(vat.gem("gold", address(gold.clip)),   0); // From clip.yank()
+        assertEq(vat.gem("gold", address(end)),         0); // From grab in end.snip() 
+        assertEq(vat.sin(address(vow)),       art1 * rate); // From grab in dog.bark()
+        assertEq(vat.vice(),                  art1 * rate); // From grab in dog.bark()
+        assertEq(vat.debt(),            tab + art1 * rate); // From frob and suck
+        assertEq(vat.dai(address(vow)),               tab); // From vat.suck()
+        assertEq(end.Art("gold") * rate,              tab); // Incrementing total Art in End
+
+        (uint ink3, uint art3) = vat.urns("gold", urn1);    // CDP after snip
+        assertEq(ink3, 10 ether);                           // All collateral returned to CDP
+        assertEq(art3 * rate, rad(16.5 ether));             // Tab amount of normalized debt transferred back into CDP
```

Comment:
I think I'd structure the assert as `assertEq(art3, tab / rate);`--you already assert on the value of tab, it's more readable, and remains correct even if `rate` doesn't evenly divide `tab`.